### PR TITLE
Support exposing ports and cross-service communication

### PR DESCRIPTION
**Pull request by gtardif**
Signed-off-by: Guillaume Tardif <guillaume.tardif@gmail.com>

**What I did**
* Add headless service for internal cross-service communication
* Services exposing ports mapped to load-balancer
* add some unit tests
* Added a sample composefile that deploys successfuly on Desktop kube cluster

**Related issue**
Fixes https://github.com/docker/compose-cli/issues/1182

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**

### CODE REVIEW

File: kube/charts/kubernetes/kube_test.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+		},
+		Spec: core.ServiceSpec{
+			Selector:  map[string]string{"com.docker.compose.service": "nginx", "com.docker.compose.project": ""},
+			ClusterIP: "None",
+			Ports: []core.ServicePort{
+				{
+					Name:       headlessName,
+					Protocol:   core.ProtocolTCP,
+					Port:       headlessPort,
+					TargetPort: intstr.IntOrString{IntVal: 55555},
```
Comment (ndeloof) [inline kube/charts/kubernetes/kube_test.go:89]: should use `headlessPort` constant here

File: kube/charts/kubernetes/kube.go [hunk 2]
```diff
@@ -33,6 +34,12 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
+const (
+	headlessPort      = 55555
```
Comment (ndeloof) [inline kube/charts/kubernetes/kube.go:38]: any reason using this specific value?

File: kube/charts/kubernetes/kube_test.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: meta.ObjectMeta{
+			Name: "nginx",
+		},
+		Spec: core.ServiceSpec{
+			Selector:  map[string]string{"com.docker.compose.service": "nginx", "com.docker.compose.project": ""},
+			ClusterIP: "None",
+			Ports: []core.ServicePort{
```
Comment (ndeloof) [inline kube/charts/kubernetes/kube_test.go:84]: This _should_  not be required with https://github.com/kubernetes/dns/issues/174

File: kube/charts/kubernetes/kube.go (hunk 2 above)
Comment (gtardif) [reply kube/charts/kubernetes/kube.go:38]: taken from compose-on-kube, no specific reason I think

File: kube/charts/kubernetes/kube_test.go (hunk 3 above)
Comment (gtardif) [reply kube/charts/kubernetes/kube_test.go:84]: `Service "web" is invalid: [spec.ports[0].port: Invalid value: 0: must be between 1 and 65535, inclusive, spec.ports[0].targetPort` when trying to deploy without this artificial port. 
versions in go.mod don't seem very old (helm 3.5.0, kube client 0.20.1, there's one minor version available, but https://github.com/kubernetes/dns/issues/174 is 3 years old, so I don't think we're behind versions fixing this...)

Comment (gtardif) [reply kube/charts/kubernetes/kube_test.go]: was probably another issue, fixed this

**Review by ndeloof — APPROVED** [review]

**Pull request merged**